pkg/util/apply/events.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2021 ScyllaDB
+
+package apply
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
+)
+
+func getTypeInfo(obj runtime.Object) (schema.GroupVersionKind, string, string) {
+	const unknown = "unknown"
+	gvk := obj.GetObjectKind().GroupVersionKind()
+	if gvk.Empty() {
+		kinds, _, _ := scheme.Scheme.ObjectKinds(obj)
+		if len(kinds) > 0 && !kinds[0].Empty() {
+			gvk = kinds[0]
+		} else {
+			gvk = schema.GroupVersionKind{Group: unknown, Version: unknown, Kind: unknown}
+		}
+	}
+
+	var name, namespace string
+	accessor, err := meta.Accessor(obj)
```

Comment:
these aren't really properties of the type (by the func name), maybe a separate function?